integration_test/Bindings/Python/DesignEntry/polynomial.py

```diff
@@ -0,0 +1,77 @@
+# RUN: %PYTHON% %s | FileCheck %s
+
+import mlir
+import circt
+
+from circt import Input, Output
+from circt import esi
+from circt.esi import types
+from circt.dialects import comb, hw
+
+
+@circt.module
+class PolynomialCompute:
+  """Module to compute ax^3 + bx^2 + cx + d for design-time coefficients"""
+
+  def __init__(self, coefficients: list[int]):
+    """coefficients is in 'd' -> 'a' order."""
+    self.__coefficients = coefficients
+
+    # Evaluate polynomial for 'x'.
+    self.x = Input(types.i32)
+    # Full result.
+    self.y = Output(types.i32)
+
+  def construct(self, x: mlir.ir.Value):
+    """Implement this module for input 'x'."""
+
+    taps: list[mlir.ir.Value] = list()
+    runningPower: list[mlir.ir.Value] = list()
+    for power, coeff in enumerate(self.__coefficients):
+      coeffVal = hw.ConstantOp(types.i32,
+                               mlir.ir.IntegerAttr.get(types.i32, coeff))
+      if power == 0:
+        newPartialSum = coeffVal.result
+      else:
+        partialSum = taps[-1]
+        if power == 1:
+          currPow = x
+        else:
+          x_power = [x for i in range(power - 1)]
+          currPow = comb.MulOp(types.i32, x_power + [runningPower[-1]]).result
+        newPartialSum = comb.AddOp(
+            types.i32,
+            [
+                partialSum,
+                comb.MulOp(types.i32, [coeffVal.result, currPow]).result
+            ]).result
+
+        runningPower.append(currPow)
+
+      taps.append(newPartialSum)
+
+    # Final output
+    self.y.set(taps[-1])
+
+
+class ExampleSystem(esi.System):
```

Comment:
Nit: does this test require ESI? For testing purposes, it might be best to just test the design entry API in a vacuum here without ESI.

Reply:
The ESI system has some really useful stuff in it so this was the quickest way to do it. We should really consider moving some of in to another dialect. But yeah, I agree the I should not include it in the integration test.